sdks/java/io/elasticsearch/src/main/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIO.java

```diff
@@ -443,8 +440,9 @@ public Read withBatchSize(long batchSize) {
 
     @Override
     public PCollection<String> expand(PBegin input) {
-      return input.apply(
-          org.apache.beam.sdk.io.Read.from(new BoundedElasticsearchSource(this, null)));
+      int backendVersion = getBackendVersion(getConnectionConfiguration());
```

Comment:
+1. There was, by the way, a ticket about checking the backend version at the latest possible moment in case the backend was not available. So I totally agree. Precisely I was looking for the right place to put this call, thanks!

Reply:
Well, the constructor is invoked right here in `expand`, so in order to delay the call until execution time you'll need to put the call elsewhere :) split() is invoked at execution time.

Reply:
Off course, silly me :)
moved the call to source.split()